design/user_data_incentives.md

```diff
@@ -0,0 +1,119 @@
+
+### Incentives
+
+There has to be a reason for a node operator to provide this service to users. If there is no payoff, then most operators could just block the ports
+and not accept incoming connections.
+
+Here is a list and analyisis of various potential reasons for offering this service
+
+#### 1. App developer
+
+Someone developed an dApp and wants to support the users, since they are directly interested in the growth of the application. It could be a company, or a DAO.
+A game developer would not want to reply to requests made by a DEX user.
+
+For this to be feasible, the enclave must be able to restrict requests to a whitelist of apps. Which is a technical feature that has to be implemented. 
+
+
+#### 2. Fees
+
+Node operators could charge fees for this service.
+
+Given that everyone is now expecting this to be a free service, this is unlikely to be something that has a chance.
+
+
+#### 3. Incentives payed by the protocol.
+
+The network (or protocol) charges fees from user when submitting transactions. This is something that users expect to pay.
+
+The Obscuro protocol is designed in such a way that it decouples the income from the costs by maintaing a buffer.
+
+We can use this designed mechanism to pay for node usage as well along with the L1 gas fees and the general incentives to follow the protocol.
+
+
+##### Measuring node usage
+
+As a proxy for a node responding to user requests, we can use a model where a node is payed a percentage of the gas fees that originated from their node. 
+
+A user that is connected to a node that doesn't respond to requests (like transaction receipts, or events), will leave that node and connect to a different node.
+
+The "Wallet extension", our client-side software can hide this complexity from the user, and switch nodes without the user noticing.
+
+This is not a perfect system, but it should be a good approximation. It turns a node operator into a service provider that makes an income based on the quality of service it provides. 
+
+It also removes the need to incentivise verifiers to break the "verifier's dilemma"
+
+
+##### Technical implementation
+
+To implement the mechanism described above, the network needs to keep track of the original node that received the transaction via RPC.
+
+Upon receiving a signed transaction from the wallet (MetaMask), the wallet extension will create a payload like this:
+
+```
+type RPCTxReceiver struct {
+  SignedTransaction []byte   // The signed transaction generated by the wallet.
+  RPCNode           Address  // The address of the node that accepted the request.
+  R, S              *big.Int // Signature by the registered viewing key of the transaction signer.
+}
+```
+
+The transaction is wrapped together with a node address, and signed with the viewing key. 
+
+A node will only accept transactions via RPC if they're addressed to it, and thus can use this payload to claim a reward. 
```

Comment:
Does this require the centralised sequencer to know the viewing keys of all users?

Reply:
this design is agnostic from how block producing is achieved.
I'll add that this wrapped transaction has to e accompanied by the signed viewin key